src/Request.php

```diff
@@ -48,11 +48,67 @@ public function getHttpVersion()
         return $this->httpVersion;
     }
 
+    /**
+     * Returns ALL headers
+     *
+     * This will return an (possibly empty) assoc array with header names as
+     * key and header values as value. The header value will be a string if
+     * there's only a single value or an array of strings if this header has
+     * multiple values.
+     *
+     * Note that this differs from the PSR-7 implementation of this method.
+     *
+     * @return array
+     */
     public function getHeaders()
     {
         return $this->headers;
     }
 
+    /**
+     * Retrieves a message header value by the given case-insensitive name.
+     *
+     * @param string $name
+     * @return string[] a list of all values for this header name or an empty array if header was not found
+     */
+    public function getHeader($name)
+    {
+        $found = array();
+
+        $name = strtolower($name);
+        foreach ($this->headers as $key => $value) {
+            if (strtolower($key) === $name) {
+                foreach((array)$value as $one) {
+                    $found []= $one;
```

Comment:
There are valid reasons for both styles IMO, but I have no real preference here, so yeah :+1: